sqlglot/parser.py

```diff
@@ -3993,9 +3990,11 @@ def _parse_generated_as_identity(
 
         return this
 
-    def _parse_inline(self) -> exp.InlineLengthColumnConstraint:
+    def _parse_inline(self) -> t.Optional[exp.InlineLengthColumnConstraint]:
         self._match_text_seq("LENGTH")
-        return self.expression(exp.InlineLengthColumnConstraint, this=self._parse_bitwise())
+        if self._match(TokenType.NUMBER):
+            return self.expression(exp.InlineLengthColumnConstraint, this=self._prev.text)
```

Comment:
What prompted the changes to this method?

Reply:
A test in identity.sql was failing ( `CREATE TABLE foo (baz DATE FORMAT 'YYYY/MM/DD' TITLE 'title' INLINE LENGTH 1 COMPRESS ('a', 'b')`) which, upon reading the [spec](https://www.ibm.com/docs/en/db2-for-zos/13?topic=statements-create-table), seemed like this was the thing to fix.